### PR TITLE
`CONSVC-2086` test: include tests in pydocstyle linting

### PR DESCRIPTION
**Pull request by Trinaa**
# Description
- Update the `Makefile` to include the `tests` directory in pydocstyle linting
- Update the `.pre-commit-config.yaml` file to including the tests directory in pydocstyle linting
- Update the `pyproject.toml` file to include test modules
- Remediate errors in `tests` directory

# Issue(s)
#124 | [CONSVC-2086](https://mozilla-hub.atlassian.net/browse/CONSVC-2086)

### CODE REVIEW

File: tests/integration/api/test_version.py [hunk 1]
```diff
@@ -30,6 +30,10 @@ def test_version(client: TestClient) -> None:
 
 
 def test_version_error(mocker: MockerFixture, client: TestClient) -> None:
+    """
```
Comment (ncloudioj) [inline tests/integration/api/test_version.py:33]: Hmm, shouldn't it be a single line w/o the linebreak here? E.g. `"""Test that...`

**Review by ncloudioj — APPROVED** [review]
This is great, glad to see test files are enforced with doc styling now. Thanks!

The only question I have is that I think we landed on using the `"""Begin of the doc..."""` when we first enabled pydocstyle, is that still the case? I am seeing both formats in this PR, looks like pydocstyle was fine with that.

File: tests/unit/middleware/test_featureflags.py [hunk 2]
```diff
@@ -14,7 +14,7 @@
 
 @pytest.fixture(name="featureflags_middleware")
 def fixture_featureflags_middleware(mocker: MockerFixture) -> FeatureFlagsMiddleware:
-    """Creates a FeatureFlagsMiddleware object for test"""
+    """Create a FeatureFlagsMiddleware object for test"""
```
Comment (taddes) [inline tests/unit/middleware/test_featureflags.py:17]: Just need to add a period here too

File: tests/unit/middleware/test_geolocation.py [hunk 3]
```diff
@@ -15,7 +15,7 @@
 
 @pytest.fixture(name="geolocation_middleware")
 def fixture_geolocation_middleware(mocker: MockerFixture) -> GeolocationMiddleware:
-    """Creates a GeolocationMiddleware object for test"""
+    """Create a GeolocationMiddleware object for test"""
```
Comment (taddes) [inline tests/unit/middleware/test_geolocation.py:18]: Period here.

File: tests/unit/middleware/test_user_agent.py [hunk 4]
```diff
@@ -14,7 +14,7 @@
 
 @pytest.fixture(name="user_agent_middleware")
 def fixture_user_agent_middleware(mocker: MockerFixture) -> UserAgentMiddleware:
-    """Creates a UserAgentMiddleware object for test"""
+    """Create a UserAgentMiddleware object for test"""
```
Comment (taddes) [inline tests/unit/middleware/test_user_agent.py:17]: Period here.

File: tests/unit/providers/test_accuweather.py [hunk 5]
```diff
@@ -360,8 +358,10 @@ async def test_invalid_location_key_forecast(
 async def test_no_client_country_or_postal_code(
     caplog: LogCaptureFixture, accuweather: Provider, geolocation: Location
 ):
-    """Test that if a client has an unknown country or postal code, that a
-    warning is logged and no suggestions are returned."""
+    """
+    Test that if a client has an unknown country or postal code, that a warning is
```
Comment (taddes) [inline tests/unit/providers/test_accuweather.py:362]: I'd say a slight grammatical shift could make it clearer, like `Test that if a client has an unknown country or postal code, a warning is logged and no suggestions are returned.` Not a big deal, just a suggestion.

File: tests/unit/test_featureflags.py [hunk 6]
```diff
@@ -1,25 +1,42 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+"""
+Unit tests for the featureflags.py module.
+
+This module depends on testing.toml
```
Comment (taddes) [inline tests/unit/test_featureflags.py:8]: Spot to add a period on line 8

**Review by taddes — APPROVED** [review]
Really great work @Trinaa , this is a big one. I know putting in tons of docstrings is tiresome, tedious and you have to put the context together on a lot of this, but the payback is huge. Thank you and also thanks from all the future developers who benefit from this clarity. I added a few suggestions, mostly just adding periods to a few docstring conclusions and one grammatical suggestion. I'll approve so as to not block you since those are minor fixes, but just tweak those and go ahead and merge!

**Comment by Trinaa** [comment]
> This is great, glad to see test files are enforced with doc styling now. Thanks!
> 
> The only question I have is that I think we landed on using the `"""Begin of the doc..."""` when we first enabled pydocstyle, is that still the case? I am seeing both formats in this PR, looks like pydocstyle was fine with that.

@ncloudioj  Does `"""Begin of the doc..."""` mean that the preference is to put the doc string on line 1? 
To look like the following?
image

**Comment by ncloudioj** [comment]
> Does """Begin of the doc...""" mean that the preference is to put the doc string on line 1?
To look like the following for example?

Not really. I was asking shall start the doc immediately after `"""` without the line break? E.g. do this

```
"""This is an example."""
``` 

rather than

```
"""
This is an example.
"""
```

**Comment by Trinaa** [comment]
@ncloudioj 

Thank you for the clarification. After doing a quick search, I believe why this isn't enforced is because we have not enabled the D212 convention.

image

Because D212 and D213 are conflicting they are disabled by default.

Furthermore, the D209 convention prevents us from keeping closing quotes in-line.

I will attempt to enable D212 and disable D209, to better enforce our preferences.

File: tests/integration/api/test_version.py (hunk 1 above)
Comment (Trinaa) [reply tests/integration/api/test_version.py]: Change applied: [3478ff3](https://github.com/mozilla-services/merino-py/pull/125/commits/3478ff3afc93cf9ee89e8ac283c7287b2ca30935)

File: tests/integration/__init__.py [hunk 7]
```diff
@@ -0,0 +1,5 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+"""Modules dedicated to integration testing the merino service"""
```
[the remark this answers is not in the document]
Comment (Trinaa) [reply tests/integration/__init__.py:5]: Change applied [f2d9ac5](https://github.com/mozilla-services/merino-py/pull/125/commits/f2d9ac506f6212e2e6d6be194e2b383291ba4663)

File: tests/integration/api/__init__.py [hunk 8]
```diff
@@ -0,0 +1,5 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+"""Modules dedicated to integration testing the merino API"""
```
[the remark this answers is not in the document]
Comment (Trinaa) [reply tests/integration/api/__init__.py]: Change applied [f2d9ac5](https://github.com/mozilla-services/merino-py/pull/125/commits/f2d9ac506f6212e2e6d6be194e2b383291ba4663)

File: tests/unit/middleware/test_featureflags.py (hunk 2 above)
Comment (Trinaa) [reply tests/unit/middleware/test_featureflags.py]: Change applied [f2d9ac5](https://github.com/mozilla-services/merino-py/pull/125/commits/f2d9ac506f6212e2e6d6be194e2b383291ba4663)

File: tests/unit/middleware/test_geolocation.py (hunk 3 above)
Comment (Trinaa) [reply tests/unit/middleware/test_geolocation.py]: Change applied [f2d9ac5](https://github.com/mozilla-services/merino-py/pull/125/commits/f2d9ac506f6212e2e6d6be194e2b383291ba4663)

File: tests/unit/middleware/test_user_agent.py (hunk 4 above)
Comment (Trinaa) [reply tests/unit/middleware/test_user_agent.py]: Change applied [f2d9ac5](https://github.com/mozilla-services/merino-py/pull/125/commits/f2d9ac506f6212e2e6d6be194e2b383291ba4663)

File: tests/unit/providers/test_accuweather.py (hunk 5 above)
Comment (Trinaa) [reply tests/unit/providers/test_accuweather.py]: Change applied [f2d9ac5](https://github.com/mozilla-services/merino-py/pull/125/commits/f2d9ac506f6212e2e6d6be194e2b383291ba4663)

File: tests/unit/test_featureflags.py (hunk 6 above)
Comment (Trinaa) [reply tests/unit/test_featureflags.py]: Change applied [f2d9ac5](https://github.com/mozilla-services/merino-py/pull/125/commits/f2d9ac506f6212e2e6d6be194e2b383291ba4663)

**Pull request merged**